### ISSUE TITLE
No return value where it is declared there should be one

### ISSUE THREAD

**Issue by wodny** [wallix/pylibssh2#3]
Patch below...

```
From edd1a5a1aa904491cdd80d301c083e70a7e05cee Mon Sep 17 00:00:00 2001
From: Marcin Szewczyk <marcin.szewczyk@wodny.org>
Date: Tue, 28 Jun 2011 22:13:39 +0200
Subject: [PATCH 1/2] Return values as declared. Added _session in
 session_methods.

---
 libssh2/session.py |   14 +++++++-------
 1 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/libssh2/session.py b/libssh2/session.py
index afed5e3..2b85dc5 100644
--- a/libssh2/session.py
+++ b/libssh2/session.py
@@ -62,7 +62,7 @@ class Session(object):
         @return: 0 on success or negative on failure
         @rtype: int
         """
-        self._session.close(reason)
+        return self._session.close(reason)

     def direct_tcpip(self, host, port, shost, sport):
         """
@@ -183,7 +183,7 @@ class Session(object):
         @return: 0 on success or negative on failure
         @rtype: int
         """
-        self._session.session_methods(method_type, pref)
+        return self._session.session_methods(method_type, pref)

     def session_methods(self):
         """
@@ -193,7 +193,7 @@ class Session(object):
         @return: dictionnary with actual method negociated
         @rtype: dict
         """
-        return self.session_methods()
+        return self._session.session_methods()

     def set_banner(self, banner=_libssh2.DEFAULT_BANNER):
         """
@@ -206,7 +206,7 @@ class Session(object):
         @return: 0 on success or negative on failure
         @rtype: int
         """
-        self._session.set_banner(banner)
+        return self._session.set_banner(banner)

     def sftp_init(self):
         """
@@ -227,7 +227,7 @@ class Session(object):
         @return: 0 on success or negative on failure
         @rtype: int
         """
-        self._session.startup(sock)
+        return self._session.startup(sock)

     def userauth_authenticated(self):
         """
@@ -249,7 +249,7 @@ class Session(object):
         methods
         @rtype: str
         """
-        self._session.userauth_list(username)
+        return self._session.userauth_list(username)

     def userauth_password(self, username, password):
         """
@@ -263,7 +263,7 @@ class Session(object):
         @return: 0 on success or negative on failure
         @rtype: int
         """
-        self._session.userauth_password(username, password)
+        return self._session.userauth_password(username, password)

     def userauth_publickey_fromfile(
             self, username, publickey, privatekey, passphrase
-- 
1.7.5.4
```

**Comment by ohe** [comment]
Corrected in pull request related to issue #5

**Comment by wodny** [comment]
What about returns for:
session_method_pref()
set_banner()
startup()

And my patch contains one more very import part:

```
-        return self.session_methods()
+        return self._session.session_methods()
```

without "_session" it falls into endless recursion.

**Comment by sbz** [comment]
Thanks wodny, it's fixed in commit 17eb29c0f8df3299e56a70bf2d089133cbd80f29
